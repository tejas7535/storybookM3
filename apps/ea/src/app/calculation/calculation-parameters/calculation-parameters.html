<!-- eslint-disable @angular-eslint/template/cyclomatic-complexity -->
<ng-container *transloco="let t; read: 'operationConditions'">
  <div class="flex w-full flex-col md:w-[660px] lg:w-[850px]">
    <div class="flex flex-1 flex-col items-center gap-10">
      <div class="flex flex-col gap-0 w-full sm:gap-6 lg:grid">
        <ea-qualtrics-info-banner></ea-qualtrics-info-banner>

        <div class="text-medium-emphasis px-1 text-h6">
          {{t('title', {bearingDesignation: bearingDesignation$ | ngrxPush } )}}
        </div>
        <div class="py-2 flex flex-col gap-y-2" *ngIf="!isProduction">
          <ea-info-banner
            [isWarning]="true"
            [infoText]="t('prototypeWarning')"
          ></ea-info-banner>
        </div>
        <div class="flex-row items-end w-full mb-4 flex md:hidden">
          <button
            (click)="onShowBasicFrequenciesDialogClick()"
            mat-button
            class="ml-auto"
            color="primary"
          >
            {{ 'basicFrequencies.showBasicFrequencies' | transloco }}
          </button>
        </div>

        <div class="flex flex-row items-center md:hidden justify-between mb-4">
          <button
            mat-fab
            extended
            color="primary"
            (click)="onShowCalculationTypesClick()"
            class="ml-auto mr-2 h-8 px-4"
          >
            <mat-icon class="!m-auto">tune</mat-icon>
          </button>
        </div>

        <!-- DEV STUFF -->
        <div
          *ngIf="isDev"
          class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4"
        >
          <div class="flex flex-col">
            <mat-slide-toggle
              [checked]="false"
              #showRawErrors
              class="text-body-1 m-2"
              color="primary"
              labelPosition="after"
            >
              Show form errors
            </mat-slide-toggle>

            <ng-container *ngIf="showRawErrors.checked">
              <pre class="bg-surface border-border border rounded-md mt-2 p-2">
Status: {{ operationConditionsForm.status }}</pre
              >

              <pre
                *ngIf="operationConditionsForm.invalid"
                class="bg-surface border-border border rounded-md mt-2 p-2"
              >
{{ formErrors$ | ngrxPush | json}}</pre
              >
            </ng-container>
          </div>

          <div class="flex flex-col">
            <mat-slide-toggle
              [checked]="false"
              #showFormValue
              class="text-body-1 m-2"
              color="primary"
              labelPosition="after"
            >
              Show form value
            </mat-slide-toggle>

            <pre
              class="bg-surface border-border border rounded-md mt-2 p-2"
              *ngIf="showFormValue.checked"
            >
{{ operationConditionsForm.value | json }}</pre
            >
          </div>
        </div>

        <!-- INPUT PARAMETERS -->
        <ng-container *ngIf="parameterTemplates">
          <!-- MANDATORY PARAMETERS -->
          <ng-container
            *ngFor="let parameterTemplate of parameterTemplates.mandatory"
            [ngTemplateOutlet]="parameterTemplate"
            [ngTemplateOutletContext]="{ t: t }"
          >
          </ng-container>

          <!-- PRESET PARAMETERS -->
          <div class="flex w-full justify-end items-center">
            <mat-slide-toggle
              [checked]="false"
              #showPresets
              class="text-body-1"
              color="primary"
              labelPosition="before"
            >
              {{ t('showPresetCalculationParameters') }}
            </mat-slide-toggle>
          </div>
          <div *ngIf="showPresets.checked" class="flex flex-col gap-4">
            <ng-container
              *ngFor="let parameterTemplate of parameterTemplates.preset"
              [ngTemplateOutlet]="parameterTemplate"
              [ngTemplateOutletContext]="{ t: t }"
            >
            </ng-container>
          </div>
        </ng-container>

        <div class="flex">
          <button
            mat-button
            color="primary"
            (click)="onResetButtonClick()"
            icon
          >
            <mat-icon>refresh</mat-icon>
            {{ t('resetInputs') }}
          </button>
        </div>
      </div>
    </div>
  </div>
</ng-container>

<!-- LOAD -->
<ng-template parameterTemplate="load" let-t="t">
  <ea-input-group [title]="t('loads')" icon="compress" [isMandatory]="true">
    <div
      class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4"
      *ngIf="loadAvailable$ | ngrxPush as loadAvailable; else loading"
    >
      <ea-input-number
        *ngIf="loadAvailable.radialLoad"
        [formControl]="operationConditionsForm.controls.load.controls.radialLoad"
        [label]="t('radialLoad')"
        [unit]="'N'"
        [tooltip]="t('tooltip.radialLoad')"
        [placeholder]="t('pleaseEnter', { label: t('radialLoad') })"
        [onlyPositive]="true"
        [customErrors]="[{name: 'anyLoad'}]"
      ></ea-input-number>
      <ea-input-number
        *ngIf="loadAvailable.axialLoad"
        [formControl]="operationConditionsForm.controls.load.controls.axialLoad"
        [label]="t('axialLoad')"
        [unit]="'N'"
        [tooltip]="t('tooltip.axialLoad')"
        [placeholder]="t('pleaseEnter', { label: t('axialLoad') })"
        [onlyPositive]="true"
        [customErrors]="[{name: 'anyLoad'}]"
      ></ea-input-number>
    </div>
  </ea-input-group>
</ng-template>

<!-- MOTION -->
<ng-template parameterTemplate="rotatingCondition" let-t="t">
  <ea-input-group
    [title]="t('title')"
    icon="animation"
    [isMandatory]="true"
    *transloco="let t; read: 'operationConditions.rotatingCondition'"
  >
    <ea-tabbed-options
      [simplifiedMobile]="true"
      *ngIf="typeOfMotionsAvailable$ | ngrxPush as typeOfMotionsAvailable else loading"
      [formControl]="operationConditionsForm.controls.rotation.controls.typeOfMotion"
    >
      <ng-template
        optionTemplate="LB_ROTATING"
        [label]="t('options.LB_ROTATING')"
        *ngIf="typeOfMotionsAvailable.includes('LB_ROTATING')"
      >
        <div class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4 w-full">
          <ea-input-number
            [formControl]="operationConditionsForm.controls.rotation.controls.rotationalSpeed"
            [label]="t('relativeRotationalSpeed')"
            [unit]="t('onePerMinute')"
            [tooltip]="t('tooltip.rotationSpeed')"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('relativeRotationalSpeed')}"
            [onlyPositive]="true"
          ></ea-input-number>
        </div>
      </ng-template>

      <ng-template
        optionTemplate="LB_OSCILLATING"
        [label]="t('options.LB_OSCILLATING')"
        *ngIf="typeOfMotionsAvailable.includes('LB_OSCILLATING')"
      >
        <div class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4 w-full">
          <ea-input-number
            [formControl]="operationConditionsForm.controls.rotation.controls.shiftFrequency"
            [label]="t('shiftFrequency')"
            [unit]="t('rpm')"
            [tooltip]="t('tooltip.shiftFrequency')"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('shiftFrequency')}"
            [onlyPositive]="true"
          ></ea-input-number>
          <ea-input-number
            [formControl]="operationConditionsForm.controls.rotation.controls.shiftAngle"
            [label]="t('shiftAngle')"
            [unit]="t('degree')"
            [tooltip]="t('tooltip.shiftAngle')"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('shiftAngle')}"
            [onlyPositive]="true"
          ></ea-input-number>
        </div>
      </ng-template>
    </ea-tabbed-options>
  </ea-input-group>
</ng-template>

<!-- LUBRICATION -->
<ng-template parameterTemplate="lubrication" let-t="t">
  <ea-input-group
    [title]="t('title')"
    svgIcon="water_drop"
    [isMandatory]="true"
    [formGroup]="operationConditionsForm.controls.lubrication"
    *transloco="let t; read: 'operationConditions.lubrication'"
  >
    <ea-tabbed-options
      formControlName="lubricationSelection"
      *ngIf="lubricationMethodsAvailable$ | ngrxPush as lubricationMethodsAvailable else loading"
    >
      <ng-template
        optionTemplate="grease"
        [label]="t('options.grease')"
        formGroupName="grease"
        [disabled]="!lubricationMethodsAvailable.grease"
        [tooltip]="lubricationMethodsAvailable.grease ? undefined : t('notAvailable')"
      >
        <!-- TYPE OF GREASE -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.typeOfGrease.label')"
          name="typeOfGrease"
          class="grow shrink basis-0"
          [description]="t('grease.typeOfGrease.description')"
        >
          <ng-container formGroupName="typeOfGrease">
            <ea-input-select
              formControlName="typeOfGrease"
              [label]="t('grease.typeOfGrease.label')"
              [options]="greases"
              [tooltip]="t('grease.typeOfGrease.tooltip')"
            ></ea-input-select>
          </ng-container>
        </ea-tabbed-suboption>

        <!-- ISO VG CLASS -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.isoVgClass.label')"
          name="isoVgClass"
          class="grow shrink basis-0"
          [description]="t('grease.isoVgClass.description')"
        >
          <ng-container formGroupName="isoVgClass">
            <ea-input-select
              formControlName="isoVgClass"
              [label]="t('grease.isoVgClass.label')"
              [options]="isoVgClasses"
              [tooltip]="t('grease.isoVgClass.tooltip')"
            ></ea-input-select>
          </ng-container>
        </ea-tabbed-suboption>

        <!-- VISCOSITY -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.viscosity.label')"
          name="viscosity"
          class="grow shrink basis-0"
          [description]="t('grease.viscosity.description')"
        >
          <ng-container formGroupName="viscosity">
            <ea-input-number
              formControlName="ny40"
              [label]="t('grease.viscosity.ny40')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
              class="min-w-max"
            ></ea-input-number>

            <ea-input-number
              formControlName="ny100"
              [label]="t('grease.viscosity.ny100')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
              class="min-w-max"
            ></ea-input-number>
          </ng-container>
        </ea-tabbed-suboption>
      </ng-template>

      <ng-template
        optionTemplate="oilBath"
        [label]="t('options.oilBath')"
        formGroupName="oilBath"
        [disabled]="!lubricationMethodsAvailable.oilBath"
        [tooltip]="lubricationMethodsAvailable.oilBath ? undefined : t('notAvailable')"
      >
        <!-- ISO VG CLASS -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.isoVgClass.label')"
          name="isoVgClass"
          class="grow shrink basis-0"
          [description]="t('grease.isoVgClass.description')"
        >
          <ng-container formGroupName="isoVgClass">
            <ea-input-select
              formControlName="isoVgClass"
              [label]="t('grease.isoVgClass.label')"
              [options]="isoVgClasses"
              [tooltip]="t('grease.isoVgClass.tooltip')"
            ></ea-input-select>
          </ng-container>
        </ea-tabbed-suboption>

        <!-- VISCOSITY -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.viscosity.label')"
          name="viscosity"
          class="grow shrink basis-0"
          [description]="t('grease.viscosity.description')"
        >
          <ng-container formGroupName="viscosity">
            <ea-input-number
              formControlName="ny40"
              [label]="t('grease.viscosity.ny40')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
            ></ea-input-number>

            <ea-input-number
              formControlName="ny100"
              [label]="t('grease.viscosity.ny100')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
            ></ea-input-number>
          </ng-container>
        </ea-tabbed-suboption>
      </ng-template>

      <ng-template
        optionTemplate="oilMist"
        [label]="t('options.oilMist')"
        formGroupName="oilMist"
        [disabled]="!lubricationMethodsAvailable.oilMist"
        [tooltip]="lubricationMethodsAvailable.oilMist ? undefined : t('notAvailable')"
      >
        <!-- ISO VG CLASS -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.isoVgClass.label')"
          name="isoVgClass"
          class="grow shrink basis-0"
          [description]="t('grease.isoVgClass.description')"
        >
          <ng-container formGroupName="isoVgClass">
            <ea-input-select
              formControlName="isoVgClass"
              [label]="t('grease.isoVgClass.label')"
              [options]="isoVgClasses"
              [tooltip]="t('grease.isoVgClass.tooltip')"
            ></ea-input-select>
          </ng-container>
        </ea-tabbed-suboption>

        <!-- VISCOSITY -->
        <ea-tabbed-suboption
          formControlName="selection"
          [label]="t('grease.viscosity.label')"
          name="viscosity"
          class="grow shrink basis-0"
          [description]="t('grease.viscosity.description')"
        >
          <ng-container formGroupName="viscosity">
            <ea-input-number
              formControlName="ny40"
              [label]="t('grease.viscosity.ny40')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
            ></ea-input-number>

            <ea-input-number
              formControlName="ny100"
              [label]="t('grease.viscosity.ny100')"
              [unit]="t('grease.viscosity.unit')"
              [onlyPositive]="true"
              [customErrors]="[{name: 'viscosity'}]"
            ></ea-input-number>
          </ng-container>
        </ea-tabbed-suboption>
      </ng-template>

      <ng-template
        optionTemplate="recirculatingOil"
        [label]="t('options.recirculatingOil')"
        formGroupName="recirculatingOil"
        [className]="'flex-col'"
        [disabled]="!lubricationMethodsAvailable.recirculatingOil"
        [tooltip]="lubricationMethodsAvailable.recirculatingOil ? undefined : t('notAvailable')"
      >
        <div class="flex flex-col md:flex-row gap-4">
          <!-- ISO VG CLASS -->
          <ea-tabbed-suboption
            formControlName="selection"
            [label]="t('grease.isoVgClass.label')"
            name="isoVgClass"
            class="grow shrink basis-0"
            [description]="t('grease.isoVgClass.description')"
          >
            <ng-container formGroupName="isoVgClass">
              <ea-input-select
                formControlName="isoVgClass"
                [label]="t('grease.isoVgClass.label')"
                [options]="isoVgClasses"
                [tooltip]="t('grease.isoVgClass.tooltip')"
              ></ea-input-select>
            </ng-container>
          </ea-tabbed-suboption>

          <!-- VISCOSITY -->
          <ea-tabbed-suboption
            formControlName="selection"
            [label]="t('grease.viscosity.label')"
            name="viscosity"
            class="grow shrink basis-0"
            [description]="t('grease.viscosity.description')"
          >
            <ng-container formGroupName="viscosity">
              <ea-input-number
                formControlName="ny40"
                [label]="t('grease.viscosity.ny40')"
                [unit]="t('grease.viscosity.unit')"
                [onlyPositive]="true"
                [customErrors]="[{name: 'viscosity'}]"
              ></ea-input-number>

              <ea-input-number
                formControlName="ny100"
                [label]="t('grease.viscosity.ny100')"
                [unit]="t('grease.viscosity.unit')"
                [onlyPositive]="true"
                [customErrors]="[{name: 'viscosity'}]"
              ></ea-input-number>
            </ng-container>
          </ea-tabbed-suboption>
        </div>

        <div class="grid grid-cols-1 mt-4 gap-0 md:grid-cols-3 md:gap-4">
          <ea-input-number
            formControlName="oilFlow"
            [label]="t('recirculatingOil.oilFlow')"
            [unit]="'l/min'"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('recirculatingOil.oilFlow')}"
            [onlyPositive]="true"
          ></ea-input-number>
          <ea-input-number
            formControlName="oilTemperatureDifference"
            [label]="t('recirculatingOil.oilTemperatureDifference')"
            [unit]="'K'"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('recirculatingOil.oilTemperatureDifference')}"
            [onlyPositive]="true"
          ></ea-input-number>
          <ea-input-number
            formControlName="externalHeatFlow"
            [label]="t('recirculatingOil.externalHeatFlow')"
            [unit]="'KW'"
            [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('recirculatingOil.externalHeatFlow')}"
            [onlyPositive]="true"
          ></ea-input-number>
        </div>
      </ng-template>
    </ea-tabbed-options>
  </ea-input-group>
</ng-template>

<!-- TEMPERATURE -->
<ng-template
  [parameterTemplate]="['operatingTemperature', 'ambientTemperature']"
>
  <ea-input-group
    [title]="t('title')"
    svgIcon="device_thermostat"
    [isMandatory]="true"
    *transloco="let t; read: 'operationConditions.temperature'"
  >
    <div class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4">
      <ea-input-number
        [formControl]="operationConditionsForm.controls.operatingTemperature"
        [label]="t('operatingTemperature.label')"
        [unit]="t('operatingTemperature.unit')"
        [tooltip]="t('operatingTemperature.tooltip')"
        [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('operatingTemperature.label')}"
        [onlyPositive]="true"
        [customErrors]="[{name: 'tmpBoundViolation'}]"
      ></ea-input-number>

      <ea-input-number
        [formControl]="operationConditionsForm.controls.ambientTemperature"
        [label]="t('ambientTemperature.label')"
        [unit]="t('ambientTemperature.unit')"
        [tooltip]="t('ambientTemperature.tooltip')"
        [placeholder]="'operationConditions.pleaseEnter' | transloco:{ label: t('ambientTemperature.label')}"
        [customErrors]="[{name: 'tmpBoundViolation'}]"
      ></ea-input-number>
    </div>
  </ea-input-group>
</ng-template>

<!-- CONTAMINATION -->
<ng-template parameterTemplate="contamination">
  <ea-input-group
    [title]="t('title')"
    icon="flip_camera_android"
    [isMandatory]="true"
    *transloco="let t; read: 'operationConditions.contamination'"
  >
    <div class="grid grid-cols-1 gap-0 md:grid-cols-2 md:gap-4">
      <ea-input-select
        [formControl]="operationConditionsForm.controls.contamination"
        [label]="t('label')"
        [options]="contaminationOptions$ | ngrxPush"
      ></ea-input-select>
    </div>
  </ea-input-group>
</ng-template>

<!-- CONDITION OF ROTATION -->
<ng-template parameterTemplate="conditionOfRotation">
  <ea-input-group
    [title]="t('title')"
    icon="flip_camera_android"
    [isMandatory]="true"
    *transloco="let t; read: 'operationConditions.conditionOfRotation'"
  >
    <ea-tabbed-options
      mobileStyleClass="bg-transparent !gap-0 pb-0"
      [formControl]="operationConditionsForm.controls.conditionOfRotation"
    >
      <ng-template
        optionTemplate="innerring"
        [label]="t('options.rotatingInnerring')"
      >
      </ng-template>

      <ng-template
        optionTemplate="outerring"
        [label]="t('options.rotatingOuterring')"
      >
      </ng-template>
    </ea-tabbed-options>
  </ea-input-group>
</ng-template>

<ng-template #loading>
  <div class="flex items-center justify-center md:mb-4">
    <mat-spinner diameter="16" /></div
></ng-template>
