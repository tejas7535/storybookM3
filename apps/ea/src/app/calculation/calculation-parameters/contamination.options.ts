import { combineLatest, map, Observable } from 'rxjs';

import { CalculationParametersOperationConditions } from '@ea/core/store/models';
import { TranslocoService } from '@ngneat/transloco';

export const getContaminationOptions = (
  translocoService: TranslocoService
): Observable<
  {
    label: string;
    value: CalculationParametersOperationConditions['contamination'];
  }[]
> =>
  combineLatest([
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.extremeCleanliness'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_EXTREM_CLEANLINESS' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.highCleanliness'
      )
      .pipe(map((label) => ({ label, value: 'LB_HIGH_CLEANLINESS' as const }))),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.standardCleanliness'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_STANDARD_CLEANLINESS' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.slightContamination'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_SLIGHT_CONTAMINATION' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.typicalContamination'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_TYPICAL_CONTAMINATION' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.heavyContamination'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_HEAVY_CONTAMINATION' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.contamination.options.veryHeavyContamination'
      )
      .pipe(
        map((label) => ({
          label,
          value: 'LB_VERY_HEAVY_CONTAMINATION' as const,
        }))
      ),
  ]);
