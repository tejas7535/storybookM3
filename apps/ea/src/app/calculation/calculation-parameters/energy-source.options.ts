import { combineLatest, Observable } from 'rxjs';

import {
  ElectricityRegionOption,
  FossilOriginOption,
} from '@ea/core/services/calculation-parameters';
import {
  ELECTRICITY_REGION_VALUE_MAPPING,
  FOSSIL_FACTORS_VALUE_MAPPING,
} from '@ea/core/services/downstream-calcululation.service.constant';
import { createTranslationObservables } from '@ea/shared/helper';
import { TranslocoService } from '@jsverse/transloco';

export const getElectricityRegionOptions = (
  translocoService: TranslocoService
): Observable<ElectricityRegionOption[]> => {
  const baseKey = 'operationConditions.energySource.electricityRegionOption.';

  const countryOptions = [...ELECTRICITY_REGION_VALUE_MAPPING.entries()].map(
    ([key, value]) => ({
      key,
      value,
    })
  );

  const observables = createTranslationObservables(
    translocoService,
    baseKey,
    countryOptions
  );

  return combineLatest(observables);
};

export const getFossilOriginOptions = (
  translocoService: TranslocoService
): Observable<FossilOriginOption[]> => {
  const baseKey = 'operationConditions.energySource.fossilOriginOptions.';

  const fossilFactors = [...FOSSIL_FACTORS_VALUE_MAPPING.entries()].map(
    ([key, value]) => ({
      key,
      value,
    })
  );

  const observables = createTranslationObservables(
    translocoService,
    baseKey,
    fossilFactors
  );

  return combineLatest(observables);
};
