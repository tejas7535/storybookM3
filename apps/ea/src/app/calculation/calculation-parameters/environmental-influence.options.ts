import { combineLatest, map, Observable } from 'rxjs';

import { CalculationParametersOperationConditions } from '@ea/core/store/models';
import { TranslocoService } from '@jsverse/transloco';

export const getEnvironmentalInfluenceOptions = (
  translocoService: TranslocoService
): Observable<
  {
    label: string;
    value: CalculationParametersOperationConditions['lubrication']['grease']['environmentalInfluence'];
  }[]
> =>
  combineLatest([
    translocoService
      .selectTranslate(
        'operationConditions.lubrication.grease.environmentalInfluence.options.lowInfluence'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_LOW_AMBIENT_INFLUENCE' as const }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.lubrication.grease.environmentalInfluence.options.averageInfluence'
      )
      .pipe(
        map((label) => ({
          label,
          value: 'LB_AVERAGE_AMBIENT_INFLUENCE' as const,
        }))
      ),
    translocoService
      .selectTranslate(
        'operationConditions.lubrication.grease.environmentalInfluence.options.heavyInfluence'
      )
      .pipe(
        map((label) => ({ label, value: 'LB_HIGH_AMBIENT_INFLUENCE' as const }))
      ),
  ]);
