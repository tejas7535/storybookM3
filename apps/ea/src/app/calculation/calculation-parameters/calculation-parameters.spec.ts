import { MatButtonModule } from '@angular/material/button';
import { MatIconTestingModule } from '@angular/material/icon/testing';
import { MatTooltipModule } from '@angular/material/tooltip';

import { resetCalculationParameters } from '@ea/core/store/actions/calculation-parameters/calculation-parameters.actions';
import { FormFieldModule } from '@ea/shared/form-field';
import {
  CALCULATION_PARAMETERS_STATE_MOCK,
  CALCULATION_RESULT_STATE_MOCK,
  PRODUCT_SELECTION_STATE_MOCK,
} from '@ea/testing/mocks';
import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { translate } from '@ngneat/transloco';
import { LetModule, PushModule } from '@ngrx/component';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { MockModule } from 'ng-mocks';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { CalculationParametersComponent } from './calculation-parameters';

describe('CalculationParametersComponent', () => {
  let component: CalculationParametersComponent;
  let spectator: Spectator<CalculationParametersComponent>;
  let store: MockStore;

  const createComponent = createComponentFactory({
    component: CalculationParametersComponent,
    imports: [
      PushModule,
      LetModule,

      // Material Modules
      MockModule(MatButtonModule),
      MatIconTestingModule,
      MockModule(MatTooltipModule),
      MockModule(FormFieldModule),

      provideTranslocoTestingModule({ en: {} }),
    ],
    providers: [
      provideMockStore({
        initialState: {
          calculationParameters: { ...CALCULATION_PARAMETERS_STATE_MOCK },
          calculationResult: { ...CALCULATION_RESULT_STATE_MOCK },
          productSelection: { ...PRODUCT_SELECTION_STATE_MOCK },
        },
      }),
      {
        provide: translate,
        useValue: jest.fn(),
      },
    ],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.debugElement.componentInstance;

    store = spectator.inject(MockStore);
    store.dispatch = jest.fn();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('ngOnDestroy', () => {
    it('should call the destroy methods', () => {
      component['destroy$'].next = jest.fn();
      component['destroy$'].complete = jest.fn();

      component.ngOnDestroy();

      expect(component['destroy$'].next).toHaveBeenCalled();
      expect(component['destroy$'].complete).toHaveBeenCalled();
    });
  });

  describe('onResetButtonClick', () => {
    it('should reset the form with initialState', () => {
      component.form.reset = jest.fn();

      component.onResetButtonClick();

      expect(store.dispatch).toHaveBeenCalledWith(resetCalculationParameters());
    });
  });

  describe('loadValidator', () => {
    it('should unset the anyLoad error if one field is > 0', () => {
      component.radialLoad.setValue(0);
      component.axialLoad.setValue(0);

      component['loadValidator']();
      expect(component.radialLoad.errors).toEqual({
        anyLoad: true,
      });

      component.radialLoad.setValue(6);
      component['loadValidator']();

      /* eslint-disable unicorn/no-null */
      expect(component.radialLoad.errors).toEqual(null);
    });

    it('should set the anyLoad error if both fields are less then 0', () => {
      component.radialLoad.setValue(0);
      component.axialLoad.setValue(0);

      component['loadValidator']();

      expect(component.radialLoad.errors).toEqual({
        anyLoad: true,
      });
      expect(component.axialLoad.errors).toEqual({
        anyLoad: true,
      });
    });
  });
});
