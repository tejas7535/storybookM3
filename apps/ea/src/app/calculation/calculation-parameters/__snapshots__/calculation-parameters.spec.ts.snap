// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationParametersComponent ngAfterViewInit should setup parameterTemplates after view init 1`] = `
Object {
  "mandatory": Array [
    undefined,
  ],
  "preset": Array [
    undefined,
  ],
}
`;
